tests/test_csv.py

```diff
@@ -0,0 +1,166 @@
+import re
+from copy import copy
+from unittest import mock
+
+import pytest
+from crate.sphinx import csv
+from crate.sphinx.csv import CSVFilterDirective
+from docutils.statemachine import StringList
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        (1, 1),
+        ("1", 1),
+        (-1, ValueError),
+        ("-1", ValueError)
+    ]
+)
+def test_non_negative_int(input, expected):
+    if expected == ValueError:
+        with pytest.raises(ValueError):
+            csv.non_negative_int(input)
+    else:
+        assert csv.non_negative_int(input) == expected
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        ("1,2,3,4", [1, 2, 3, 4]),
+        ("a,2,3,4", ValueError),
+        ("-1,2,3,4", ValueError),
+        ("0,-2,3,4", ValueError),
+    ]
+)
+def test_non_negative_int_list(input, expected):
+    if expected == ValueError:
+        with pytest.raises(ValueError):
+            csv.non_negative_int_list(input)
+    else:
+        assert csv.non_negative_int_list(input) == expected
+
+
+directive_params = ['Company,Contact,Country,Attend?'], None, 'source'
+
+directive = CSVFilterDirective(
+    None, None, None, None, None, None, None, None, None)
+tmp_directive_options = {
+    'header': 'Company,Contact,Country,Attend?',
+    'delim': '\t',
+    'file': 'example.csv',
+    'included_cols': [0, 1, 2]
+}
+
+
+def test_csv_filter_exclude_directive():
+    directive_options = copy(tmp_directive_options)
+    directive_options['exclude'] = {3: '(?i)Y\\w*'}
+    directive.options = directive_options
+
+    with mock.patch.object(directive, '_apply_filters') as apply_filters:
+        rows, max_cols = directive.parse_csv_data_into_rows(*directive_params)
+        args = apply_filters.call_args
+        assert args[0][2] is None
+        assert args[0][3] == {3: re.compile('(?i)Y\\w*', re.IGNORECASE)}
+
+
+def test_csv_filter_include_directive():
+    directive_options = copy(tmp_directive_options)
+    directive_options['include'] = {3: '(?i)Y\\w*'}
+    directive.options = directive_options
+
+    with mock.patch.object(directive, '_apply_filters') as apply_filters:
+        rows, max_cols = directive.parse_csv_data_into_rows(*directive_params)
+        args = apply_filters.call_args
+        assert args[0][2] == {3: re.compile('(?i)Y\\w*', re.IGNORECASE)}
+        assert args[0][3] is None
+
+
+def test_csv_filter_include_exclude_directive():
+    directive_options = copy(tmp_directive_options)
+    directive_options['include'] = {3: '(?i)Y\\w*'}
```

Comment:
Let's use different values for `['include']` and `['exclude']` to be able to assert they've been passed correctly.